cmd/local.go

```diff
@@ -0,0 +1,233 @@
+	// Override tracing configuration if enableTracing flag is set
+	if enableTracing {
+		conf.ExternalServices.Tracing.Enabled = true
+		config.Set(conf)
+	}
+
+	// Need a separate "port-forward to prom option" because you can specify an external prometheus URL
+	// in the config file and that should not be overridden by the port-forwarding.
+	if conf.ExternalServices.Prometheus.Enabled && portForwardToPromFlag {
+		if err := portForwardToProm(ctx, cf.GetSAHomeClusterClient(), conf); err != nil {
+			log.Warningf("Unable to setup port forwarding to prom pods: %s\t Disabling prometheus.", err)
+			conf.ExternalServices.Prometheus.Enabled = false
+			config.Set(conf)
+		}
+	}
+	// Port forward to Tracing if enabled and flag is set
+	if conf.ExternalServices.Tracing.Enabled && portForwardToTracingFlag {
+		if err := portForwardToTracing(ctx, cf.GetSAHomeClusterClient(), conf); err != nil {
+			log.Warningf("Unable to setup port forwarding to tracing pods: %s", err)
+		}
+	}
+	// Port forward to Grafana if enabled and flag is set
+	if conf.ExternalServices.Grafana.Enabled && portForwardToGrafanaFlag {
+		if err := portForwardToGrafana(ctx, cf.GetSAHomeClusterClient(), conf); err != nil {
+			log.Warningf("Unable to setup port forwarding to grafana pods: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func openDefaultBrowser(ctx context.Context, conf *config.Config) error {
+	kialiURL := fmt.Sprintf("http://localhost:%d", conf.Server.Port)
+	var cmd string
+	switch runtime.GOOS {
+	// TODO: handle windows and mac.
+	default:
+		cmd = "xdg-open"
+	}
+	return exec.CommandContext(ctx, cmd, kialiURL).Start()
+}
+
+// portForwardToPod sets up port forwarding to a pod identified by namespace and labelSelector.
+// It returns the allocated local port that was used for port forwarding.
+func portForwardToPod(ctx context.Context, localClient kubernetes.ClientInterface, namespace, labelSelector string, targetPort string) (int, error) {
+	localPort := httputil.Pool.GetFreePort()
+
+	pods, err := localClient.Kube().CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return 0, err
+	}
+
+	if len(pods.Items) == 0 {
+		return 0, fmt.Errorf("no pod found in %s namespace with label selector %s", namespace, labelSelector)
+	}
+
+	pod := pods.Items[0]
+	log.Infof("Port forwarding to %s pod", pod.Name)
+	pf, err := httputil.NewPortForwarder(
+		localClient.Kube().CoreV1().RESTClient(),
+		localClient.ClusterInfo().ClientConfig,
+		pod.Namespace,
+		pod.Name,
+		"localhost",
+		fmt.Sprintf("%d:%s", localPort, targetPort),
+		io.Discard,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	if err := pf.Start(); err != nil {
+		return 0, err
+	}
+	go func() {
+		defer pf.Stop()
+		<-ctx.Done()
+	}()
+
+	return localPort, nil
+}
+
+func portForwardToTracing(ctx context.Context, localClient kubernetes.ClientInterface, conf *config.Config) error {
+	localPort, err := portForwardToPod(ctx, localClient, "", "app=jaeger", "16686")
```

Comment:
This and those below are hardcoding the Istio demo profile things, right?  Is that going to be sufficient?

Reply:
No I'll add flags that allow you to override these settings